localstack-core/localstack/testing/aws/util.py

```diff
@@ -190,6 +190,7 @@ def base_aws_session() -> boto3.Session:
     session = boto3.Session(
         aws_access_key_id=TEST_AWS_ACCESS_KEY_ID,
         aws_secret_access_key=TEST_AWS_SECRET_ACCESS_KEY,
+        region_name=TEST_AWS_REGION_NAME,
```

Comment:
It is by design that these fixtures only consider the credentials and not the region. One must use `aws_client_factory` fixture to construct a client associated with a region.

Reply:
Ah, interesting, why is that necessary? Why shouldn't it default to the region we have configured in the config?
The problem is that this fixture is also used by `aws_http_client_factory`, where we then do not take the `TEST_AWS_REGION_NAME` into account otherwise.

Reply:
It may be a good idea to keep sessions not bound to a region. If the `base_aws_session()` uses `TEST_AWS_REGION_NAME`, one might assume `secondary_aws_session()` uses the `SECONDARY_TEST_AWS_REGION_NAME`.

Consider updating `aws_http_client_factory` to take into account the `TEST_AWS_REGION_NAME` as a fallback, in line with how `base_aws_client_factory` does it

Reply:
Thanks, I moved setting the region to the `aws_http_client_factory`. Let me know what you think :)